### ISSUE TITLE
rake fails with Redmine 4.0 / Rails 5 due to alias_method_chain

### ISSUE THREAD

**Issue by MikeDitumTps** [akiko-pusu/redmine_issue_templates#205]
## Summary
Trying to install the plugin on Redmine 4.0 / Rails 5 fails

## Description

I'm trying to use this plugin in Redmine 4.0 and it's failing when I run...
`bundle exec rake redmine:plugins NAME=redmine_issue_templates RAILS_ENV=production`

Initially it failed with...
```
[root@redmine redmine]#     bundle exec rake redmine:plugins NAME=redmine_issue_templates RAILS_ENV=production
rake aborted!
NameError: uninitialized constant Rake::TestTask
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/tasks/test.rake:7:in `block in <top (required)>'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/tasks/test.rake:1:in `<top (required)>'
/opt/redmine/lib/tasks/redmine.rake:190:in `block in <top (required)>'
/opt/redmine/lib/tasks/redmine.rake:190:in `each'
/opt/redmine/lib/tasks/redmine.rake:190:in `<top (required)>'

```
But I managed to fix that by adding 'require "rake/testtask.rb"' at the top of lib/tasks/test.rake

Now it fails with...
```
[root@redmine redmine]# bundle exec rake redmine:plugins NAME=redmine_issue_templates RAILS_ENV=production
rake aborted!
NoMethodError: undefined method `alias_method_chain' for ProjectsHelper:Module
Did you mean?  alias_method
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/issue_templates/projects_helper_patch.rb:8:in `block in included'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/issue_templates/projects_helper_patch.rb:7:in `class_eval'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/issue_templates/projects_helper_patch.rb:7:in `included'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/init.rb:53:in `include'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/init.rb:53:in `block (2 levels) in <top (required)>'
```

It appears that Rails 5 has removed support for alias_method_chain (see https://www.justinweiss.com/articles/rails-5-module-number-prepend-and-the-end-of-alias-method-chain/) but my understanding of Ruby is not good enough to fix it.

## Environment

- Redmine version: 4.0.0.stable.17771
- Installed plugins:   
  redmine_agile                  1.4.7
  redmine_checklists             3.1.14
  redmine_mentions               0.0.1
  redmineup_tags                 2.0.4
  redmine_issue_templates           0.2.1
- Ruby version: 2.5.3-p105
- OS Platform: Centos 7
- Database: MySQL 5.7
- Rails: 5.2.2

## Visual Proof / Screenshot

```
[root@redmine redmine]# bundle exec rake redmine:plugins NAME=redmine_issue_templates RAILS_ENV=production
rake aborted!
NoMethodError: undefined method `alias_method_chain' for ProjectsHelper:Module
Did you mean?  alias_method
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/issue_templates/projects_helper_patch.rb:8:in `block in included'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/issue_templates/projects_helper_patch.rb:7:in `class_eval'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/lib/issue_templates/projects_helper_patch.rb:7:in `included'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/init.rb:53:in `include'
/opt/redmine/plugins/redmine_issue_templates-0.2.1/init.rb:53:in `block (2 levels) in <top (required)>'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:426:in `instance_exec'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:426:in `block in make_lambda'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:198:in `block (2 levels) in halting'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:606:in `block (2 levels) in default_terminator'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:605:in `catch'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:605:in `block in default_terminator'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:199:in `block in halting'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:513:in `block in invoke_before'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:513:in `each'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:513:in `invoke_before'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/callbacks.rb:131:in `run_callbacks'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/reloader.rb:89:in `prepare!'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/application/finisher.rb:63:in `block in <module:Finisher>'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/initializable.rb:32:in `instance_exec'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/initializable.rb:32:in `run'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/initializable.rb:61:in `block in run_initializers'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/initializable.rb:60:in `run_initializers'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/application.rb:361:in `initialize!'
/opt/redmine/config/environment.rb:14:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/dependencies.rb:291:in `require'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/dependencies.rb:291:in `block in require'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/dependencies.rb:257:in `load_dependency'
/usr/local/rvm/gems/ruby-2.5.3/gems/activesupport-5.2.2/lib/active_support/dependencies.rb:291:in `require'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/application.rb:337:in `require_environment!'
/usr/local/rvm/gems/ruby-2.5.3/gems/railties-5.2.2/lib/rails/application.rb:520:in `block in run_tasks_blocks'
/opt/redmine/lib/tasks/redmine.rake:57:in `block (2 levels) in <top (required)>'
/usr/local/rvm/gems/ruby-2.5.3/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.5.3/bin/ruby_executable_hooks:24:in `eval'
/usr/local/rvm/gems/ruby-2.5.3/bin/ruby_executable_hooks:24:in `<main>'
Tasks: TOP => redmine:plugins:migrate => environment
(See full trace by running task with --trace)
```

## Expected Results
rake task would complete

## Actual Results
Fails

## Workaround
None for alias_method_chain issue.

**Comment by MikeDitumTps** [comment]
I've tested the support-rails5 branch and that compiles correctly and seems to be working so it looks like it's fixed just needs to be included into master.

**Comment by akiko-pusu** [comment]
@MikeDitumTps 
Thanks for trying my plugin!
Yes, I’ve not released newer version to support Redline 4.0.
And support-rails5 branch is exactly WIP branch for Redmine4.0 but I stoped development for Redmine4.0 and prioritized the maintenance of the Redmine 3.x. 

This is because I could not make sure when Redmine4.0 would be released....
So, support-rails5 branch is not applied some bug fix or implementation of master branch since last spring.

Please wait for a while. I might be able to release new version soon within 1 week or so.

**Comment by akiko-pusu** [comment]
Fixed and done refactoring. Now merged into develop branch.

**Comment by MrsKensington** [comment]
I've just tested the develop branch and it is working correctly. I've run some quick tests on my test redmine setup and it seems to be behaving in the same way as 0.1.3 on redmine 3.x.

Thanks for the quick response!

**Comment by MikeDitumTps** [comment]
I've just tested the develop branch and it is working correctly. I've run some quick tests on my test redmine setup and it seems to be behaving in the same way as 0.1.3 on redmine 3.x.

Thanks for the quick response!

**Comment by akiko-pusu** [comment]
I’ve just released v0.3.0. Thank you so much your request and cheering!
